kedro-telemetry/kedro_telemetry/plugin.py

```diff
@@ -231,18 +247,24 @@ def after_catalog_created(self, catalog):
         default_pipeline = pipelines.get("__default__")  # __default__
         user_uuid = _get_or_create_uuid()
 
-        project_properties = _get_project_properties(
-            user_uuid, self.project_path / PYPROJECT_CONFIG_NAME
-        )
+        if not self.event_properties:
+            event_properties = _get_project_properties(
+                user_uuid, self.project_path / PYPROJECT_CONFIG_NAME
+            )
+        else:
+            event_properties = self.event_properties
 
-        project_statistics_properties = _format_project_statistics_data(
-            project_properties, catalog, default_pipeline, pipelines
-        )
-        _send_heap_event(
-            event_name="Kedro Project Statistics",
-            identity=user_uuid,
-            properties=project_statistics_properties,
+        enriched_properties = _format_project_statistics_data(
+            event_properties, catalog, default_pipeline, pipelines
         )
+        self.event_properties = enriched_properties
+        if not self._sent:
```

Comment:
I believe we don't need that check here. The logic should be that `enriched_properties` should be sent in any case. It is our responsibility to ensure the event is not sent again in the `after_command_run()` hook later.

Reply:
Yeah I think you are right, since we only have two hooks `after_catalog_created` and `after_command_run` that will send event, and this is the earliest one so in reality there shouldn't be case this is set `True`.